### PR TITLE
[fix](cluster key) fix compaction and schema change

### PR DESCRIPTION
**Pull request by mymeiyi**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:
1. fix compaction: 
    the overlapping is set according to the min max key of priamry key index.  When compaction for mow tables with cluster keys, if multi segments is not overlapping, but the cluster key may be overlapping, so should open all segments.
    the segment is overlap for cluster key tables.
3. fix schema change:
    the external sorting use a non vertical compaction to sort the rowsets generated by internal sort, but cluster key tables does not support non vertical compaction.

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by hello-stephen** [comment]
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

**Review by zhannngchen — DISMISSED** [review]
LGTM

File: be/src/olap/schema_change.cpp [hunk 1]
```diff
@@ -728,8 +732,27 @@ Status VBaseSchemaChangeWithSorting::_external_sorting(vector<RowsetSharedPtr>&
     }
 
     Merger::Statistics stats;
-    RETURN_IF_ERROR(Merger::vmerge_rowsets(new_tablet, ReaderType::READER_ALTER_TABLE,
-                                           *new_tablet_schema, rs_readers, rowset_writer, &stats));
+    if (!new_tablet_schema->cluster_key_uids().empty()) {
+        // schema change read rowsets with delete bitmap, so there should be no duplicated keys
+        // RETURN_IF_ERROR(Compaction::update_delete_bitmap());
```
Comment (zhannngchen) [inline be/src/olap/schema_change.cpp:737]: remove this line

**Comment by mymeiyi** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by zhannngchen — APPROVED** [review]
LGTM

**Pull request merged**